### PR TITLE
Rebirth (former working title "New Game+") feature

### PR DESCRIPTION
**Pull request by SabreCat**
habitrpg-shared implementation of the "Rebirth" feature!

See https://github.com/HabitRPG/habitrpg/pull/2111 for full description of the feature. These commits include sprite assets and shared code (user ops).

Also fixes an incidentally discovered bug where "Fortify" would zero out the prices of the user's custom Rewards.

### CODE REVIEW

**Comment by SabreCat** [comment]
Man, next time around I need to ponder a different means of installing a branched habitrpg-shared than pushing commits with every change. This makes it look like I'm trying to artificially inflate my GitHub stats xD

File: dist/habitrpg-shared.js [hunk 1]
```diff
@@ -10715,7 +10715,7 @@ var process=require("__browserify_process");(function() {
         newHistory.push({
           date: moment(group[0].date).toDate(),
           value: _.reduce(group, (function(m, obj) {
-            return m + api.value;
+            return m + obj.value;
```
Comment (SabreCat) [inline dist/habitrpg-shared.js:10718]: Not sure what this is. Came along with @lefnire's grunt commit.

Comment (lefnire) [inline dist/habitrpg-shared.js:10718]: yeah that one's vital, pay no heed. a @lefnire face-palm, @djuretic saved the day

Comment (djuretic) [inline dist/habitrpg-shared.js:10718]: Actually, it was the unit test that @lefnire wrote that helped me to diagnose the problem.

Comment (SabreCat) [inline dist/habitrpg-shared.js:10718]: :heart: for unit tests!

**Comment by SabreCat** [comment]
It's like a bonus Christmas! :gift:

**Pull request merged**